randovania/lib/migration_lib.py

```diff
@@ -1,36 +1,48 @@
 from __future__ import annotations
 
 import copy
+import functools
 import typing
-from collections.abc import Callable
+
+if typing.TYPE_CHECKING:
+    from randovania.game.game_enum import RandovaniaGame
 
 
 class UnsupportedVersion(ValueError):
     pass
 
 
-Migrations = typing.Sequence[Callable[[dict], dict] | None]
+class Migration(typing.Protocol):
+    def __call__(self, __data: dict) -> None: ...
+
+
+class GameMigration(typing.Protocol):
+    def __call__(self, __data: dict, game: RandovaniaGame) -> None: ...
```

Comment:
this is a cool thing with protocols. this protocol describes a callable (which returns `None`) with exactly two arguments, where the first argument is a `dict` and can have any name (due to starting with a double underscore), while the second argument is a `RandovaniaGame` and must be named `game`